src/adaptations/device-layer/nRF5/ConfigurationManagerImpl.cpp

```diff
@@ -77,6 +79,13 @@ WEAVE_ERROR ConfigurationManagerImpl::_Init()
     return err;
 }
 
+WEAVE_ERROR ConfigurationManagerImpl::_GetPrimary802154MACAddress(uint8_t * buf)
+{
+    const otExtAddress *extendedAddr = otLinkGetExtendedAddress(ThreadStackMgrImpl().OTInstance());
```

Comment:
While this approach works, it could be implemented more generically, such that GetPrimary802154MACAddress() automatically works on any platform that incorporates OpenThread.  E.g.: 

- GenericConfigurationManagerImpl<>::_GetPrimary802154MACAddress() could be rewritten to conditionally forward the call to a new interface method declared on ThreadStackManager; e.g.: ThreadStackMgr().GetPrimary802154MACAdddres().  The forwarding behavior would be conditional on WEAVE_DEVICE_CONFIG_ENABLE_THREAD being set; if not set, it would return WEAVE_DEVICE_ERROR_CONFIG_NOT_FOUND as it does not.

- GenericThreadStackManagerImpl_OpenThread<> would then provide a default implementation of the new ThreadStackManager method that works by calling otLinkGetExtendedAddress().

Reply:
Done, I've uploaded a new changeset